### ISSUE TITLE
Column browser

### ISSUE THREAD

**Issue by ishmaelen** [heyarne/airsonic-ui#35]
Awesome that somebody is finally modernizing the Airsonic UI. Kudos to you Arne! 

Have you considered a column browser? 
Something like this (see attached pic - based on MM4 browser) would be awesome in my POV. 

It should read locations/nodes for the tree, tags for the genre/artist/album (and more), files for the tracks window below with user set choice of columns showing tags for files. Users should even be allowed to resize the different parts as they please. 

I guess its mostly useful for people with large collections or a wide variation of tags.

![columnbrowser](https://user-images.githubusercontent.com/2997171/49529303-0962b400-f8b6-11e8-8113-f1953478ce05.png)

**Comment by ishmaelen** [comment]
I use airsonic today (besides a whole bunch of other music-related apps) so a modernization of a UI that has the same sort of view is more than good enough for me!

As I stated my guess is that only old nerds like myself with huge collections or people with a large amount of different tags for a "high-level" classification (I have tagged my collection to about 650 different genres) would appreciate a column browser. And yes, you would probably have to hold a seperate list of items/info to display it quick enough and avoid all that overhead. The upside for me with a column browser is quick browsing and easily identify tracks that needs tag-fixing (and I could perhaps hack this UI to run beets for me in the background and fix tags automagically from a MusicBrainz DB on the same server for speed reasons..)

I do like the coverArtService.js in current Airsonic to e.g. fix/add/change artwork though so please keep something similar around. Should have a Google search besides Last.FM and URL specifier. Preferably other sources like MediaBrainz,Discogs etc. as well. 

Have a look at the newest MM5 beta instead to get some feel instead of the old MM4. 
They are currently doing a revamp of their UI as well. 

Smart playlists are cool, sort of like the different "cloud" implementations you see around and I love filters (tags and more).

Closing the issue - thanks for the feedback. Please keep up the great work!

**Comment by heyarne** [comment]
> I do like the coverArtService.js in current Airsonic to e.g. fix/add/change artwork though so please keep something similar around.

I like the idea but it never worked for me. I plan on keeping it around but it depends on work being done on the server as there currently is not API endpoint exposing that functionality.

> Should have a Google search besides Last.FM and URL specifier. Preferably other sources like MediaBrainz,Discogs etc. as well.

More links would be cool, yes. My collection yields some links to MusicBrainz; this really depends on whether your music has a MusicBrainz ID in its ID3 tags. Other things like searching for the artist on DuckDuckGo or Google (maybe make that configurable) would be cool, to, I agree.

Again, thanks for the input and kind words!

**Comment by ishmaelen** [comment]
> > I do like the coverArtService.js in current Airsonic to e.g. fix/add/change artwork though so please keep something similar around.
> 
> I like the idea but it never worked for me. I plan on keeping it around but it depends on work being done on the server as there currently is not API endpoint exposing that functionality.
> 
> > Should have a Google search besides Last.FM and URL specifier. Preferably other sources like MediaBrainz,Discogs etc. as well.
> 
> More links would be cool, yes. My collection yields some links to MusicBrainz; this really depends on whether your music has a MusicBrainz ID in its ID3 tags. Other things like searching for the artist on DuckDuckGo or Google (maybe make that configurable) would be cool, to, I agree. 
> 
> Again, thanks for the input and kind words!

The MusicBrainz IDs if not found in ID3 tags can be searched from the acoustic fingerprinting (see https://beets.io & https://acoustid.org) and is achievable using e.g. chromaprint.

Just a tip you might find useful for your own tagging-adventures.

**Comment by heyarne** [comment]
Thanks for the tip. The UI uses what the airsonic server hands it; trying to incorporate this information automatically in the future might be worth investigating. I myself use [MusicBrainz Picard](https://picard.musicbrainz.org/) to clean up my tags at the moment; I used beets before but I got lazy when tagging a kind of big collection and ended up having wrong tags on a lot of tracks which I ended up having to clean manually. Picard uses the same fingerprinting stuff under the hood I guess; I was just suggesting that those text might not be present in your tags when you don't see the links – or maybe airsonic didn't catch them, in which case you found a bug.
